### PR TITLE
feat: support iOS LivePhoto backup

### LINKED ISSUE #160: Feature: iOS Live Photos
iPhone Live photos are converted into images. If Immich is used as a backup, the old status cannot be restored.

### PR DESCRIPTION
**Pull request by alextran1502**
This PR will enable Live Photos (LP) backup and retrieval support.

We will be working with two parts of the Live Photos asset: the image and the video.

### To Do

- [x] Add a table to store LP's video information
- ~~[ ] Add an option DTO property(ies) to flag a video belonging to an LP and which LP it belongs to.~~
- [x] Update `photo_manager` dependency to support LP's feature
- [x] Implement upload LP type on mobile client
- [x] Record the LP's video info to the respective table in the same `uploadFile` endpoint for all asset-upload-related tasks.
- [x] Conditionally display LP asset on mobile app with an option to play the video in looping mode
- [x] Conditionally display LP asset on the web with an option to play the video part in looping mode
- [x] Merge the image and video into one file when downloaded to the phone.
- [x] Downloaded as two separate files if downloaded from the web
- [x] Delete photo and motion record in the database and on disk
- [ ] Make LP a searchable term [optional]

### Hover
https://user-images.githubusercontent.com/27055614/202816753-0b0eb201-1747-4fef-8b31-ce380a851bbc.mov

### Play

https://user-images.githubusercontent.com/27055614/202816825-63ab1ca1-8b08-4bdc-8dff-2dc802c847e7.mov

## Mobile Interaction

https://user-images.githubusercontent.com/27055614/202816953-42321a01-8906-481f-ac5e-7e5601ed1774.mov

### CODE REVIEW

File: server/libs/database/src/entities/asset.entity.ts [hunk 1]
```diff
@@ -51,6 +51,12 @@ export class AssetEntity {
   @Column({ type: 'varchar', nullable: true })
   duration!: string | null;
 
+  @Column({ type: 'boolean', default: true })
+  isVisible!: boolean;
+
+  @Column({ type: 'uuid', nullable: true })
+  livePhotoVideoId!: string | null;
```
Comment (jrasm91) [inline server/libs/database/src/entities/asset.entity.ts:58]: What do you think about adding a one to one relationship to the live video and loading it as part of the asset list / get by id? Or, do you not use any data from it except the asset id to load the video?

**Review by jrasm91 — APPROVED** [review]
I only looked through the code, but no major issues from me. 

I think there might be some potential room for improvement around the upload endpoint. It seems like there is a lot of really similar code repeated for both the normal asset upload and the optional video portion that could probably be simplified/streamlined.

File: server/libs/database/src/entities/asset.entity.ts (hunk 1 above)
Comment (alextran1502) [reply server/libs/database/src/entities/asset.entity.ts:58]: That is correct, so far it has been the main usage, to use the id to load the video. The secondary usage is for downloading which can also be done purely by assetId

Comment (jrasm91) [reply server/libs/database/src/entities/asset.entity.ts:58]: Sounds good.

**Pull request merged**